### PR TITLE
[Cart] Drop cart validation

### PR DESCRIPTION
**Pull request by lchrusciel**

### CODE REVIEW

File: spec/Command/DropCartSpec.php [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    }
+
+    function it_has_order_token()
+    {
+        $this->orderToken()->shouldReturn('ORDERTOKEN');
+    }
+
+    function it_throws_an_exception_if_order_token_is_not_a_string()
+    {
+        $this->beConstructedWith(new \stdClass(), 'example@customer.com');
```
Comment (pamil) [inline spec/Command/DropCartSpec.php]: What's the second argument?

File: spec/Handler/DropCartHandlerSpec.php [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Core\Repository\OrderRepositoryInterface;
+use PhpSpec\ObjectBehavior;
+use Sylius\ShopApiPlugin\Command\DropCart;
+use Sylius\ShopApiPlugin\Handler\DropCartHandler;
+
+final class DropCartHandlerSpec extends ObjectBehavior
+{
+    function let(
+        OrderRepositoryInterface $cartRepository
```
Comment (pamil) [inline spec/Handler/DropCartHandlerSpec.php]: Guess it can be put in a single line.

File: src/Handler/DropCartHandler.php [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    /**
+     * @var OrderRepositoryInterface
+     */
+    private $cartRepository;
+
+    /**
+     * @param OrderRepositoryInterface $cartRepository
+     */
+    public function __construct(
+        OrderRepositoryInterface $cartRepository
```
Comment (pamil) [inline src/Handler/DropCartHandler.php]: Same here.

File: tests/Request/DropCartRequestTest.php [hunk 4]
```diff
@@ -0,0 +1,20 @@
+<?php
+
+namespace Tests\Sylius\ShopApiPlugin\Request;
+
+use Sylius\ShopApiPlugin\Command\DropCart;
+use Sylius\ShopApiPlugin\Request\DropCartRequest;
+use Symfony\Component\HttpFoundation\Request;
+
+final class DropCartRequestTest extends \PHPUnit_Framework_TestCase
```
Comment (pamil) [inline tests/Request/DropCartRequestTest.php]: We can use `PHPUnit\Framework\TestCase` instead to allow for forward compatibility.

File: tests/Controller/CartDropCartApiTest.php [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+            ]), Address::createFromArray([
+            'firstName' => 'Sherlock',
+            'lastName' => 'Holmes',
+            'city' => 'London',
+            'street' => 'Baker Street 221b',
+            'countryCode' => 'GB',
+            'postcode' => 'NWB',
+            'provinceName' => 'Greater London',
+        ])
+        ));
```
Comment (pamil) [inline tests/Controller/CartDropCartApiTest.php]: Some indentation logic broken here.

**Comment by pjedrzejewski** [comment]
👍

**Pull request merged**